cupyx/scatter.py

```diff
@@ -54,3 +54,77 @@ def scatter_add(a, slices, value):
 
     """
     a.scatter_add(slices, value)
+
+
+def scatter_max(a, slices, value):
+    """Stores a maximum value of elements specified by indices to an array.
+
+    It stores a maximum value of elements of ``value`` specified by ``slices``
+    to ``a``. If all of the indices target different locations, the operation
+    of :func:`scatter_max` is equivalent to
+    ``a[slices] = cupy.maximum(a[slices], value)``.
+    If there are multiple elements targeting the same location,
+    :func:`scatter_max` stores the maximum of all of these values to the given
+    index of the array.
+
+    Note that just like an array indexing, negative indices are interpreted as
+    counting from the end of an array.
+
+    Example
+    -------
+    >>> import numpy
+    >>> import cupy
+    >>> a = cupy.zeros((6,), dtype=numpy.float32)
+    >>> i = cupy.array([1, 0, 1, 2])
+    >>> v = cupy.array([1., 2., 3., -1.])
+    >>> cupyx.scatter_max(a, i, v);
+    >>> a
+    array([2., 3., 0., 0., 0., 0.], dtype=float32)
```

Comment:
Probably I am mistaken but shouldn't a be ([2,3,-1, ...])?

Reply:
Thank you for your suggestion.
Exactly, in ``scatter_max``, the initial value of `a` is also taken into account. I revised the documents.
I also wrote that the specification of ``scatter_max`` is almost the same as ``numpy.maximum.at``. I'd be glad if you could check the documents.